addons/docs/src/frameworks/vue/sourceDecorator.test.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint no-underscore-dangle: ["error", { "allow": ["_vnode"] }] */
+
+import { ComponentOptions } from 'vue';
+import Vue from 'vue/dist/vue';
+import { vnodeToString } from './sourceDecorator';
+
+expect.addSnapshotSerializer({
+  print: (val: any) => val,
+  test: (val) => typeof val === 'string',
+});
+
+const getVNode = (Component: ComponentOptions<any, any, any>) => {
+  const vm = new Vue({
+    render(h: (c: any) => unknown) {
+      return h(Component);
+    },
+  }).$mount();
+
+  return vm.$children[0]._vnode;
+};
+
+describe('vnodeToString', () => {
+  it('basic', () => {
+    expect(
+      vnodeToString(
+        getVNode({
+          template: `<button>Button</button>`,
+        })
+      )
+    ).toMatchInlineSnapshot(`<button >Button</button>`);
+  });
+
+  it('attributes', () => {
+    const MyComponent: ComponentOptions<any, any, any> = {
+      props: ['propA', 'propB', 'propC', 'propD'],
+      template: '<div/>',
+    };
+
+    expect(
+      vnodeToString(
+        getVNode({
+          components: { MyComponent },
+          data(): { props: Record<string, any> } {
+            return {
+              props: {
+                propA: 'propA',
+                propB: 1,
+                propC: null,
+                propD: {
+                  foo: 'bar',
+                },
+              },
+            };
+          },
+          template: `<my-component v-bind="props"/>`,
+        })
+      )
+    ).toMatchInlineSnapshot(
+      `<my-component :propD='{"foo":"bar"}' :propC="null" :propB="1" propA="propA"/>`
+    );
+  });
+
+  it('children', () => {
+    expect(
+      vnodeToString(
+        getVNode({
+          template: `
+          <div>
+            <form>
+              <button>Button</button>
+            </form>
+          </div>`,
+        })
+      )
+    ).toMatchInlineSnapshot(`<div ><form ><button >Button</button></form></div>`);
```

Comment:
Anyway to get rid of the extra spaces? E.g. `<div >` => `<div>`? Or is that standard in Vue?

Reply:
Yeah, it's ugly but will be erased before emitting by Prettier. Should we remove it before formatting?

Reply:
Aha didn't realize this was the pre-prettier output. If prettier fixes, that's fine by me.

Reply:
I wonder how I can write a story to achieve this effect!